### PR TITLE
Added NavBar and Context to the root Component #11

### PR DESCRIPTION
**Pull request by MokhammadDaniyal**
**What is done?**
Added NavBar and Context to the root Component 
Configured Auth flow - Navbar will be shown only to the signed in users

**further refactoring**
Fetch user from the server and store user info in App.js 

In order to simulate user not being signed in change 
`useState(true)` to `useState(null)` in App.js on line 16.
**What to avoid**

### CODE REVIEW

**Comment by abhishekchohan** [comment]
Good Morning @MokhammadDaniyal , 
Please attach some screenshots if possible.
Thanks

**Review by abhishekchohan — COMMENTED** [review]
Seems like our boilerplate with fully functional navbar for whole project is ready and just need to work on specific pages. Great work @MokhammadDaniyal   👍 

@skassam21  Will provide the detailed review. For me, the code look fine.
See you in standup..
- Abhi

File: client/src/App.js [hunk 1]
```diff
@@ -1,26 +1,53 @@
...
   return (
     <MuiThemeProvider theme={theme}>
       <BrowserRouter>
         <Switch>
-          <Route exact path="/" component={SignUp} />
-          <Route path="/signup" component={SignUp} />
-          <Route path="/signin" component={SignIn} />
-          <Route path="/onboard" component={OnboardingExperience} />
-          <Route path="*" component={SignUp} />
+          {!user && AuthStack()}
```
Comment (skassam21) [inline client/src/App.js]: Naturally these would be functional components, so you shouldn't call them, instead you should use them like components:
<AuthStack /> 

Same thing for DefaultStack below

File: client/src/components/navigation/NavbarProfile.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  return (
+    <div class={classes.profileContainer}>
+      <img
+        src={
+          "https://cdn.business2community.com/wp-content/uploads/2017/08/blank-profile-picture-973460_640.png"
```
Comment (skassam21) [inline client/src/components/navigation/NavbarProfile.js]: let's maybe make this a constant variable at the top of the file

**Review by skassam21 — APPROVED** [review]
Looks good - one small comment, please change before you merge!

**Pull request merged**